lib/src/socket_connector.dart

```diff
@@ -172,34 +179,41 @@ class SocketConnector {
             }
           }
           try {
-            if (side.farSide!.state == SideState.open) {
-              side.farSide!.sink.add(data);
-            } else {
-              throw StateError(
-                  'Will not write to side ${side.farSide!.name} as its state is ${side.farSide!.state}');
+            side.farSide!.sink.add(data);
+            if (side.farSide!.sink is Socket) {
+              side.farSide!.sent += data.length;
+              if (side.state == SideState.closed &&
+                  side.rcvd == side.farSide!.sent) {
+                _closeSide(side.farSide!);
+              }
```

Comment:
This code is here because sometimes we will have a transformer for writing to the far side, in which case the check on sent / rcvd happens in the transformer's listen block above; but if there is no transformer (i.e. the farSide's sink is a Socket) then we do the same check here